### PR TITLE
Bug fix/cleanup aql collection handling

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Cleanup the handling of `_collection` inside the `aql::Collection` class.
`_collection` is now set in the constructor, and not from the outside via `setCollection(...)`. `_collection` will only be populated for "real" collections and shards though. `aql::Collection` objects can also be created for views and for non-existing collections. In these cases it is not possible to populate `_collection`, at all.
Whenever `getCollection()` is called, `_collection` must be a non-nullptr or an assertion will be triggered. In non-maintainer mode an exception will be thrown.
`setCollection(...)` has been removed entirely.

- [x] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests, such as *all AQL tests (cluster, single server), gtests*.

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/9857/

### CODE REVIEW

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/9889/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/9903/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/9927/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/9991/

File: arangod/Aql/AqlTransaction.cpp [hunk 1]
```diff
@@ -79,28 +79,19 @@ AqlTransaction::AqlTransaction(
...
-  res = addCollection(cid, collection->name(), collection->accessType());
-
-  if (res.ok() && col != nullptr) {
-    collection->setCollection(col);
+  if (cid == 0) {
+    cid = resolver()->getCollectionId(collection->name());
   }
-
-  return res;
+  return addCollection(cid, collection->name(), collection->accessType());
```
Comment (graetzer) [inline arangod/Aql/AqlTransaction.cpp:96]: why can we not pass through the collection object

File: arangod/Aql/AqlTransaction.cpp [hunk 2]
```diff
@@ -79,28 +79,19 @@ AqlTransaction::AqlTransaction(
...
-
-  if (col != nullptr) {
-    cid = col->id();
-  } else {
-    cid = resolver()->getCollectionId(collection->name());
+  if (!_state->isCoordinator()) {
+    // TODO: check if we still need this resolving here, or if we can 
+    // reuse existing data from the collection object (which will be
+    // empty for views, which probably needs to be handled here)
+    auto col = resolver()->getCollection(collection->name());
```
[the remark this answers is not in the document]
Comment (jsteemann) [reply arangod/Aql/AqlTransaction.cpp]: Fixed it

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/10005/

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**